src/coreclr/jit/instr.cpp

```diff
@@ -1096,19 +1193,68 @@ void CodeGen::inst_RV_TT_IV(instruction ins, emitAttr attr, regNumber reg1, GenT
     }
 }
 
+//------------------------------------------------------------------------
+// IsEmbeddedBroadcastEnabled: determine if embedded broadcast can be enabled
+//
+// Arguments:
+//    ins       -- The instruction being emitted
+//    op        -- The second operand of the instruction.
+//
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+bool CodeGenInterface::IsEmbeddedBroadcastEnabled(instruction ins, GenTree* op)
+{
+    if (!GetEmitter()->UseEvexEncoding())
+    {
+        return false;
+    }
+    // need to check if the datatype is EB compatible, say 32-, 64-bit.
+    insFlags flags                    = instInfo[ins];
+    bool     IsEmbBroadcastCompatible = (flags & INS_Flags_EmbeddedBroadcastSupported) != 0;
+    if (!IsEmbBroadcastCompatible)
+    {
+        return false;
+    }
+
+    insFlags inputSize = static_cast<insFlags>((CodeGenInterface::instInfo[ins] & Input_Mask));
+
+    // Embedded broadcast can be applied when operands are in the following forms.
+    // 1. Broadcast -> CreateScalar -> LCL_VAR/CNS
```

Comment:
I think this comment is slightly out of date. It can apply to `Broadcast -> ContainedNode`, where `ContainedNode` can be something like `IND`, `LCL_VAR`, `LCL_FLD`, `LCL_VAR_ADDR`, `LCL_FLD_ADDR`, `VEC_CNS`, etc.

Basically anything that involves broadcasting from memory.

-- A couple of these might also have been removed or tweaked based on changes @SingleAccretion has recently done.

Reply:
We can also simplify this quite a bit by doing an early check for `if (!op->isContained() || !op->OperIsHWIntrinsic()) { return false; }`

as that will cut out any places we don't have containment and any non-hwintrinsic nodes.